R/cube-collapse-dimensions.R

```diff
@@ -0,0 +1,117 @@
+dimSums <- function (x, margin = NULL) {
+    # ensure that we have sensible margin input
+    selecteds <- is.selectedDimension(x@dims)
+    check_margins(margin, selecteds)
+    
+    # ensure that the cube is a counts cube
+    only_count_cube(x)
+    
+    # translate from user-cube margins to real-cube margins and establish the
+    # two groups of margins: those to collapse and those to keep
+    margins_to_collapse <- user2realMargin(margin, cube = x)
+    margins_to_keep <- setdiff(seq_along(x@dims), margins_to_collapse)
```

Comment:
I don't think you need this argument. You can just send `margins_to_collapse` to the internal function and it can do `off_margins <- setdiff(seq_along(dim(array_in), margins_to_collapse)`.

Reply:
Yup yup, I've removed the argument to `collapse_dims()` though I do use both `margins_to_keep` and `margins_to_collapse` in `dimSums` a few places each so I've kept their setting here.

In looking at this I also realized that with the name change, what margin refers to (the margins to keep or the margins to collapse) was at odds with how `rowSums` and `colSums` worked, so I've actually flipped it — which in hindsight I think is actually the much much more intuitive version answering the question: "I have this cube and I want to get a cube with only dimension X in it." much easier than having to figure out what the opposite set of margins is.